### ISSUE TITLE
Add owned Programs?

### ISSUE THREAD

**Issue by Shadow53** [rigetti/quil-rs#229]
`src/program/mod.rs` implements `Add` for `&Program` as such:

```rust
impl<'a, 'b> ops::Add<&'b Program> for &'a Program {
    type Output = Program;

    fn add(self, rhs: &'b Program) -> Program {
        let mut new = self.clone();
        new.calibrations.extend(rhs.calibrations.clone());
        new.frames.merge(rhs.frames.clone());
        new.memory_regions.extend(rhs.memory_regions.clone());
        new.waveforms.extend(rhs.waveforms.clone());
        new.instructions.extend(rhs.instructions.clone());
        new
    }
}
```

Was there a specific reason for this? Seems like it would be more efficient to implement adding two owned values and require them to be cloned at the call side instead, if necessary.

**Comment by MarquessV** [comment]
This was originally motivated by needing to expose an __add__ method for pyQuil, where the programs are stored references. There's no reason I can think of to not clone them at the call site instead, so I think you're good to make this change.

**Comment by MarquessV** [comment]
Closed via #249
